lib/inspec/profile_context.rb

```diff
@@ -25,7 +25,7 @@ def initialize(profile_id, backend, conf)
       @profile_id = profile_id
       @backend = backend
       @conf = conf.dup
-      @profile_name = @conf['profile'].profile_name || @profile_id if @conf['profile']
+      @profile_name = @conf.has_key?('profile') ? @conf['profile'].profile_name : @profile_id
```

Comment:
I think I'm a bit old school in saying this, but I think I prefer:
```ruby
@profile_name = @conf['profile'] ? @conf['profile'].profile_name : @profile_id
```
over your version... literally just pulling that `if` condition down to a ternary. But yours is actually a bit more polite and correct than mine.

Reply:
This took me a while because of lack of coffee, but this is basically `(a = b) if c` but really should be `a = (b if c)`.

Reply:
Right, that || if was throwing me too. I'll update. Thanks!